src/controller.js

```diff
@@ -63,131 +39,111 @@ function buildData(dataset, mainRect, dpr) {
           subRect.y += font.lineHeight + padding * 2;
           subRect.h -= font.lineHeight + padding * 2;
         }
-        ret.push(...recur(gidx + 1, rasterizeRect(subRect, dpr), sq.g, sq.s));
+        ret.push(...recur(gidx + 1, subRect, sq.g, sq.s));
       });
     }
     return ret;
   }
 
-  if (!tree.length && dataset.data.length) {
-    tree = dataset.tree = dataset.data;
-  }
-
   return glen
     ? recur(0, mainRect)
-    : squarify(tree, mainRect, key, dpr);
-}
-
-function getMinMax(data, useTree) {
-  const vMax = useTree ? 1 : maxValue(data);
-  const vMin = useTree ? 0 : minValue(data, vMax);
-  return {vMin, vMax};
-}
-
-function getArea({xScale, yScale}, data, rtl, useTree) {
-  const {vMin, vMax} = getMinMax(data, useTree);
-  const xMin = xScale.getPixelForValue(0);
-  const xMax = xScale.getPixelForValue(1);
-  const yMin = yScale.getPixelForValue(vMin);
-  const yMax = yScale.getPixelForValue(vMax);
-  return {x: xMin, y: yMax, w: xMax - xMin, h: yMin - yMax, rtl};
+    : squarify(tree, mainRect, key);
 }
 
 export default class TreemapController extends DatasetController {
   constructor(chart, datasetIndex) {
     super(chart, datasetIndex);
 
-    this._rect = undefined;
-    this._key = undefined;
     this._groups = undefined;
-    this._useTree = undefined;
+    this._key = undefined;
+    this._rect = undefined;
+    this._rtl = undefined;
   }
 
   initialize() {
     this.enableOptionSharing = true;
     super.initialize();
   }
 
-  /**
-   * @todo: Remove with https://github.com/kurkle/chartjs-chart-treemap/issues/137
-   */
-  updateRangeFromParsed(range, scale) {
-    if (range.updated) {
+  getMinMax(scale) {
+    return {
+      min: scale.min || 0,
+      max: scale.max || (scale.axis === 'x' ? scale.right - scale.left : scale.bottom - scale.top)
+    };
+  }
+
+  configure() {
+    super.configure();
+    const {xScale, yScale} = this.getMeta();
+    if (!xScale || !yScale) {
```

Comment:
In my test case, locally, where I was testing the zoom, xScale and yScale are undefined here, therefore also this._rect.

Reply:
Yes, turns out configure is only called once if the canvas is initially hidden or has zero size (which happens in the documentation for example).

Reply:
I think I need to use `linkScales` instead

Reply:
The scales, in `linkScales`, don't have any dimension. The dimension is calculated during the `scale.update()`  therefore we should manage the size after `scale.update()`, therefore in the `controller.update()`, as was before, I guess.

Reply:
I see also a possible issue setting scale.min/max because this could overlap the values set by the plugin

Reply:
Sorry but I was busy in other stuff. What I meant is that when the zoom is "reset", the controller cannot recognize it.

Reply:
Everything working in my local testing now, what do you think?

Reply:
Yes, it's working locally as well!! Thank you

Reply:
I have a couple of issues related to hoverCaption ane labelsMultilineFonts tolerance but doesn't matter. The rest seems working.

Reply:
2.1.3 or 2.2.0?

Reply:
going with 2.1.3